docs/developers/building-open-mpi.rst

```diff
@@ -1,9 +1,87 @@
 Building Open MPI
 =================
 
+General
+-------
+
 Once you have run ``autogen.pl`` successfully, you can configure and
 build Open MPI just like end users do with official distribution Open
 MPI tarballs.
 
 See the :doc:`general "Install Open MPI" documentation for more
 details. </installing-open-mpi/index>`
+
+Building Against External OpenPMIx / PRRTE
+------------------------------------------
+
+One thing that developers and/or packagers may need to do is to build
+Open MPI against an external OpenPMIx and/or PRRTE source tree (i.e.,
+an OpenPMIx and/or PRRTE installation that was not built from the
+embedded copies inside the Open MPI source tree / Git submodules).
+
+With regards to :doc:`Open MPI's required dependent libraries
+</installing-open-mpi/required-support-libraries>` (Hwloc, Libevent,
+OpenPMIx, and PRRTE), it generally is simplest to build Open MPI in
+one of two ways:
+
+#. Build and use all the **internal** copies of Open MPI's required
+   dependent libraries.
+
+   * Specifically: use the Hwloc, Libevent, OpenPMIx, and PRRTE source
+     trees that are bundled in with Open MPI's source code.
+
+#. Build and use all **external** copies of Open MPI's required
+   dependent libraries.
+
+   * Specifically: ignore the Hwloc, Libevent, OpenPMIx, and PRRTE source
+     trees that are bundled in with Open MPI's source code, and,
+     instead, compile and link Open MPI against already-installed
+     versions of these libraries.
+
+Other variations are possible, but can get tricky and complicated with
+subtle linker consequences, and are therefore not recommended.
+
+Some facts that are relevant to know when building against an external
+OpenPMIx / PRRTE:
+
+1. Open MPI, OpenPMIx, and PRRTE must all be built against the
+   **same** installation of Hwloc and Libevent.  Meaning:
+
+   * Assumedly the external OpenPMIx and PRRTE were built against
+     external Hwloc and Libevent.  Open MPI **must** compile and link
+     against the **same** Hwloc and Libevent that the external
+     OpenPMIx and PRRTE were built against.
+
+     .. admonition:: Critical
+        :class: Danger
+
+        Open MPI, OpenPMIx, and PRRTE must all use the same Hwloc and
+        Libevent libraries at run time (e.g., they must all resolve to
+        the same run-time loadable libraries at run time).
```

Comment:
Not totally correct - PRRTE doesn't have to use the same as OMPI never loads "libprrte", so there is no potential for confusion.

Reply:
By transitive property, though, isn't this true?  OMPI must use the same hwloc + libevent as PMIX, and PRTE must use the same hwloc + libevent as PMIx, so therefore don't they all have to be the same?

Reply:
you can have two versions of PMIx installed.  I don't think this is super common for a developer, but given how Slurm is packaged, may be more common there.